### PR TITLE
feat: typescript

### PR DESCRIPTION
**Pull request by 11reed**
Read #3 for more information

### CODE REVIEW

File: src/types/DefaultProps.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    backgroundColor: string;
+    burgerColor: string;
+    menuColor: string;
+}
+
+export type BurgerMenuType = IDefaultProps;
+export type BurgerButtonType = IDefaultProps;
+export type SideMenuType = IDefaultProps;
+*/
+export type IDefaultProps = {}
```
Comment (jorisBarkema) [inline src/types/DefaultProps.ts]: Is this still used anywhere?

File: package.json [hunk 2]
```diff
@@ -2,15 +2,16 @@
   "name": "svelte-burger-menu",
   "version": "1.0.5",
```
Comment (jorisBarkema) [inline package.json]: This would be breaking change so its needs a major version increase
```suggestion
  "version": "2.0.0",
```

**Review by jorisBarkema — CHANGES_REQUESTED** [review]
Thanks for the contribution, I have to say I don't have a lot of experience with typescript and even less with svelte-typescript configuration so I'll mostly trust you on the correctness of that part.

File: src/types/DefaultProps.ts (hunk 1 above)
Comment (11reed) [reply src/types/DefaultProps.ts]: No, it isn't. I placed types/DefaultProps.ts in .gitignore for now, until later on if we want to add more type safety. There really isn't any need until/if we add more features to this library.

File: package.json (hunk 2 above)
Comment (11reed) [reply package.json]: Done, although it will still need to be published to NPM.

**Review by jorisBarkema — APPROVED** [review]

**Pull request merged**